### ISSUE TITLE
top cpu usage

### ISSUE THREAD

**Issue by sasik007** [SilverAzide/Gadgets#116]
great addition but I have a big request, add to top cpu so that it displays the number of processes running, I really want it.
it would be best if you made such gadgets with taskmgr, I like to monitor the number of processes in the background

**Comment by SilverAzide** [comment]
OK, it is easy enough to get this value.  But where to show it?  The gadget is completely filled; there's no space left to show it.  The only options I can think of are to add space below the title, or show the process count where the title is when you move the mouse over the skin.  I'm not a fan of the first option because it would cause the skin to be mismatched with the others in this group.

**Comment by sasik007** [comment]
![Zrzut ekranu 2024-12-26 113539](https://github.com/user-attachments/assets/0c024101-db11-4980-8d4b-e80497007665)
 mayby here?
These gadgets are very nice, but is it not possible to make them load the processor less?

**Comment by SilverAzide** [comment]
No, there's no room there.  The CPU make/model goes there and extends all the way to the right side.

As far as making them load the processor less, that would be partly up to you.  I'm typically running 10+ Gadgets on my system and the CPU rarely goes above 0.5%.  If I only use the CPU Meter, even with all options enabled, my usage barely even registers, not more than 0.1%.

I've done as much optimization as possible, but you can do more if you want.  For example, only do a minimum of monitoring top processes, like you did here and only show the top 1.  Or turn it off completely.  Each process being monitored requires 4 measures at a minimum, so the fewer the better.  Turn off the memory composition feature, as that requires another dozen measures.  The worst offender is Drives Meter, if you are using it.  Configure the Drives Meter to monitor the minimum set of drives for your PC.  Each drive requires at least 16 measures, so use the Settings to disable as many as you can (16 measures x 26 drives is a lot).  So the fewer features you enable, the less work Rainmeter needs to do.

Also, ensure you have enabled hardware acceleration in the Rainmeter options.

**Comment by sasik007** [comment]
the most burdensome gadget is from monitoring the battery, I don't know why, and it's a pity because it's useful. and what, are you going to add this amount of processes somewhere?

**Comment by SilverAzide** [comment]
Really?  The Battery Meter is causing the impact?  Hm!  That's strange, since that is a very lightweight skin.  However, make sure you have UNchecked the "Use Automatic Failover" option.  Using this option **will** incur a performance hit, as noted in the info tip.  Otherwise, there are only a dozen measures in this skin, so it should not register nearly any impact.

As far as the count of processes, I'm currently not planning to add the count since there is no place to show it.  If you want to add it yourself, it is very simple, as it only requires one measure to get the value and one meter to display it.  Let me know and I can post the code you'd need.

**Comment by SilverAzide** [comment]
No response from OP.  If you want to add this yourself, let me know and I'll re-open the ticket and post some instructions.
